File: testers/avro/java-async/generated-and-checked-in/com/example/events/consumer/InvoiceHandler.java
package com.example.events.consumer;

import com.example.events.Invoice;
import com.example.events.header.StandardHeaders;
import java.util.concurrent.CompletableFuture;

/** Handler interface for invoice topic events */
public interface InvoiceHandler {
  /** Handle a message from the topic */
  CompletableFuture<Void> handle(String key, Invoice value, StandardHeaders headers);
}

File: testers/avro/kotlin/generated-and-checked-in/com/example/events/consumer/InvoiceHandler.kt
package com.example.events.consumer

import com.example.events.Invoice
import com.example.events.header.StandardHeaders

/** Handler interface for invoice topic events */
interface InvoiceHandler {
  /** Handle a message from the topic */
  abstract fun handle(
    key: kotlin.String,
    value: Invoice,
    headers: StandardHeaders
  )
}